openwisp_controller/config/static/config/js/widget.js

```diff
@@ -731,3 +754,61 @@ JSONEditor.defaults.themes.django = JSONEditor.AbstractTheme.extend({
         link.appendChild(image);
     }
 });
+
+JSONEditor.defaults.editors.multiple.prototype.$each = function (obj, callback) {
+    if (!obj || typeof obj !== "object") {
+        return;
+    }
+    var i;
+    if (Array.isArray(obj) || (typeof obj.length === 'number' && obj.length > 0 && (obj.length - 1) in obj)) {
+        for (i = 0; i < obj.length; i++) {
+            if (callback(i, obj[i]) === false) {
+                return;
+            }
+        }
+    } else {
+        if (Object.keys) {
+            var keys = Object.keys(obj);
+            for (i = 0; i < keys.length; i++) {
+                if (callback(keys[i], obj[keys[i]]) === false) {
+                    return;
+                }
+            }
+        } else {
+            for (i in obj) {
+                if (!obj.hasOwnProperty(i)) {
+                    continue;
+                }
+                if (callback(i, obj[i]) === false) {
+                    return;
+                }
+            }
+        }
+    }
+};
+
+// Override setValue method to allow using variables for fields with maxLength
+JSONEditor.defaults.editors.multiple.prototype.setValue = function (val, initial) {
+    // Determine type by getting the first one that validates
+    var self = this;
+    this.$each(this.validators, function (i, validator) {
+        if ((val) && typeof val === 'object') {
+            Object.entries(val).forEach(function (entry) {
+                if (typeof entry[1] === 'string' && entry[1].indexOf('{{') > -1) {
+                    delete validator.schema.properties[entry[0]];
```

Comment:
can you double check it this deletion is permanent or not?

Reply:
After looking into the source code of the library I found out that this deletion is indeed permanent. In https://github.com/openwisp/openwisp-controller/pull/410/commits/ba8fea3044f977a7bbc8b796d073da1dfb7f53e6, I have added code to restore validators to their original state after the validation is done. This means the validators are modified and restored everytime `setValue`  is called.